File: Algoritmalar/kepler_20170811/kepler_H.m
% ˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜˜
function F = kepler_H(e, M)


error = 1.e-8;
%...Starting value for F:
F = M;

ratio = 1;
while abs(ratio) > error
ratio = (e*sinh(F) - F - M)/(e*cosh(F) - 1);
F = F - ratio;
end
% ˜˜

File: Algoritmalar/Python/Kepler_H.py
import math
import sys
import os

def Kepler_H(e, M):
    
    error = 10**-8

    F = M
        
    ratio = 1
    
    while ratio > error:
        ratio = (e*math.sinh(F) - F - M) / (e*math.cosh(F) - 1)
        F = F - ratio
   
    return F
    
    



def main():

     print (Kepler_H(2.7696, 40.69))

if __name__ == "__main__":
    main()


